source/agora/test/Base.d

```diff
@@ -1363,14 +1420,20 @@ public enum NetworkTopology
 /// Node / Network / Quorum configuration for use with makeTestNetwork
 public struct TestConf
 {
+    /// Verbose test logging
+    bool verbose_test_logging = false;
+
     /// Network topology to use
     NetworkTopology topology = NetworkTopology.FullyConnected;
 
     /// Extra blocks to generate in addition to the genesis block
     size_t extra_blocks = 0;
 
-    /// Number of validator nodes to instantiate
-    size_t validators = 4;
+    /// Number of validator nodes that are to be enrolled in test setup
+    size_t validators = 6;
+
+    /// Number of validator nodes that are enrolled in test `GenesisBlock`
+    size_t genesis_validators = 6;
```

Comment:
I guess your plan is to eventually remove `genesis_validators` ?

One question that arose is: Do we want to allow settings validators to < 6, meaning forcing some validators to not re-enroll. And if so, which validators ?
I think if we did, it would make more sense to expose something on `TestAPI`. Which means we could only ever have a value here which would be >= 6, in which case I recommend to go with `extra_validators`.

Does this make sense ?

Reply:
I was thinking maybe this genesis_validators will be set to 4 and hardcoded. It will mean regenerating the test `GenesisBlock` though with only 4 enrolled validators. Not sure if this will be a big impact on other tests though?

Reply:
It should be set to 6, the same as the test genesis block.

Reply:
Why would we need to regenerate the test `GenesisBlock` though ?

Reply:
I think there was some confusion w.r.t. the task.

Previously we had tests which:
- spawned less than 6 genesis validator nodes
- spawned more than 6 genesis validator nodes

The ones with less than 6 have mostly been fixed in a previous PR. What's left is to fix the ones that generate more than 6 genesis validator nodes. Once that is fixed we can get rid of the `makeGenesisBlock` in `Test.d` and just use the already hardcoded test genesis block.

Reply:
I thought starting from 4 enrolled validators was mentioned at some point but I can stick with 6 as minimum and update any tests which still use less than 6 if needed.

Reply:
I have removed the call to makeGenesisBlock and am using the test GenesisBlock now. The test is actually failing at the moment but at least this shows the intention of use. Always start from 6 validators in the Genesis block and add extra when the validators in `Test_conf` is overridden with value more than 6. Less than 6 will trigger an assertion failure.